chef/cookbooks/crowbar/recipes/crowbar-upgrade.rb

```diff
@@ -92,6 +92,13 @@
     end
   end
 
+  # Remove existing upgrade directory. If it exists, it would contain
+  # indications of upgrade scripts runs from previous upgrade.
+  # If they'd stay, script wouldn't be executed in current upgrade
+  directory "/var/lib/crowbar/upgrade" do
```

Comment:
where is this directory created?

Reply:
It's created directly from the upgrade bash scripts, see e.g. https://github.com/crowbar/crowbar-core/blob/master/chef/cookbooks/crowbar/templates/default/crowbar-chef-upgraded.sh.erb#L21